sky/backends/backend_utils.py

```diff
@@ -1001,6 +1001,40 @@ def get_timestamp_from_run_timestamp(run_timestamp: str) -> float:
         run_timestamp.partition('-')[2], '%Y-%m-%d-%H-%M-%S-%f').timestamp()
 
 
+def count_healthy_nodes_from_ray(output,
+                                 is_local_cloud: bool = False
+                                ) -> Tuple[int, int]:
+    """Count the number of healthy nodes from the output of `ray status`."""
+
+    def get_ready_nodes(pattern, output, local=False):
+        result = pattern.findall(output)
+        # On-prem/local case is handled differently.
+        # `ray status` produces different output for local case, and
+        # we poll for number of nodes launched instead of counting for
+        # head and number of worker nodes separately (it is impossible
+        # to distinguish between head and worker node for local case).
+        if local:
+            # In the local case, ready_workers mean the total number
+            # of nodes launched, including head.
+            return len(result)
+        if len(result) == 0:
+            return 0
+        assert len(result) == 1, result
+        return int(result[0])
+
+    if is_local_cloud:
+        ready_workers = get_ready_nodes(_LAUNCHED_LOCAL_WORKER_PATTERN,
+                                        output,
+                                        local=True)
+    else:
+        ready_workers = get_ready_nodes(_LAUNCHED_WORKER_PATTERN,
+                                        output,
+                                        local=False)
+    ready_head = get_ready_nodes(_LAUNCHED_HEAD_PATTERN, output)
```

Comment:
nit
```suggestion
    def get_ready_nodes(pattern, output):
        result = pattern.findall(output)
        # On-prem/local case is handled differently.
        # `ray status` produces different output for local case, and
        # we poll for number of nodes launched instead of counting for
        # head and number of worker nodes separately (it is impossible
        # to distinguish between head and worker node for local case).
        if is_local_cloud:
            # In the local case, ready_workers mean the total number
            # of nodes launched, including head.
            return len(result)
        if len(result) == 0:
            return 0
        assert len(result) == 1, result
        return int(result[0])

    if is_local_cloud:
        ready_head = 0
        ready_workers = get_ready_nodes(_LAUNCHED_LOCAL_WORKER_PATTERN, output)
    else:
        ready_head = get_ready_nodes(_LAUNCHED_HEAD_PATTERN, output)
        ready_workers = get_ready_nodes(_LAUNCHED_WORKER_PATTERN, output)
```

Reply:
Added!